src/lib/message_filters.py

```diff
@@ -21,3 +28,30 @@ def filter_empty_messages(messages, message_keys):
     @staticmethod
     def filter_time_range(messages, time_key, start_time, end_time):
         return [td for td in messages if start_time <= isoparse(td.get(time_key)) <= end_time]
+
+    @staticmethod
+    def filter_noise_other_project(messages):
+        """
+        Filters a list of  RQA and followup survey messages which have been labelled as Noise_Other_Project(NOP).
+
+        :param messages: List of message objects to filter.
+        :type messages: list of TracedData
+        :return: Filtered list.
+        :rtype: list of TracedData
+        """
+        not_noise = []
+        for plan in PipelineConfiguration.RQA_CODING_PLANS + PipelineConfiguration.FOLLOW_UP_CODING_PLANS:
+            for td in messages:
+                if plan.binary_coded_field is not None and td[plan.binary_coded_field]["CodeID"] != plan.code_scheme.get_code_with_control_code(Codes.NOISE_OTHER_PROJECT) and \
+                        td[plan.binary_coded_field]["Checked"]:
```

Comment:
Do you need to check this field? Thought that the pipeline would set "NR" if not checked?